addons/hr_expense/models/hr_expense.py

```diff
@@ -406,7 +406,7 @@ def _compute_price_unit(self):
             if expense.state not in {'draft', 'reported'}:
                 continue
             product_id = expense.product_id
-            if product_id and expense.product_has_cost and not expense.nb_attachment:
+            if product_id and expense.product_has_cost and not expense._needs_price_unit_recalculation():
```

Comment:
```suggestion
            if expense._needs_price_unit_recalculation():
```
Might as well put all the conditions in the dedicated method :nerd_face: